code/datums/components/style/style.dm

```diff
@@ -0,0 +1,496 @@
+	var/datum/component/projectile_parry/projectile_parry
+	/// What rank, minimum, the user needs to be to hotswap items
+	var/hotswap_rank = STYLE_BRUTAL
+	/// If this is multitooled, making it make funny noises on the user's rank going up
+	var/multitooled = FALSE
+	/// A static list of lists of all the possible sounds to play when multitooled, in numerical order
+	var/static/list/rankup_sounds = list(
+		list(
+			'sound/items/style/combo_dull1.ogg',
+			'sound/items/style/combo_dull2.ogg',
+			'sound/items/style/combo_dull3.ogg',
+		),
+		list(
+			'sound/items/style/combo_cool1.ogg',
+			'sound/items/style/combo_cool2.ogg',
+			'sound/items/style/combo_cool3.ogg',
+		),
+		list(
+			'sound/items/style/combo_brutal1.ogg',
+			'sound/items/style/combo_brutal2.ogg',
+			'sound/items/style/combo_brutal3.ogg',
+		),
+		list(
+			'sound/items/style/combo_absolute1.ogg',
+			'sound/items/style/combo_absolute2.ogg',
+			'sound/items/style/combo_absolute3.ogg',
+		),
+		list(
+			'sound/items/style/combo_spaced1.ogg',
+			'sound/items/style/combo_spaced2.ogg',
+		),
+	)
+
+
+/datum/component/style/Initialize(multitooled = FALSE)
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	var/mob/mob_parent = parent
+
+	meter = new()
+	meter_image = new()
+	meter.vis_contents += meter_image
+	meter_image.add_filter("meter_mask", 1, list(type = "alpha", icon = icon('icons/hud/style_meter.dmi', "style_meter"), flags = MASK_INVERSE))
+	meter.update_appearance()
+	meter_image.update_appearance()
+
+	update_screen()
+
+	if(mob_parent.hud_used)
+		mob_parent.hud_used.static_inventory += meter
+		mob_parent.hud_used.show_hud(mob_parent.hud_used.hud_version)
+
+	START_PROCESSING(SSdcs, src)
+
+	if(multitooled)
+		src.multitooled = multitooled
+
+	RegisterSignal(src, COMSIG_ATOM_TOOL_ACT(TOOL_MULTITOOL), PROC_REF(on_parent_multitool))
+
+/datum/component/style/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_ITEM_AFTERATTACK, PROC_REF(hotswap))
+	RegisterSignal(parent, COMSIG_MOB_MINED, PROC_REF(on_mine))
+	RegisterSignal(parent, COMSIG_MOB_APPLY_DAMAGE, PROC_REF(on_take_damage))
+	RegisterSignal(parent, COMSIG_MOB_EMOTED("flip"), PROC_REF(on_flip))
+	RegisterSignal(parent, COMSIG_MOB_EMOTED("spin"), PROC_REF(on_spin))
+	RegisterSignal(parent, COMSIG_MOB_ITEM_ATTACK, PROC_REF(on_attack))
+	RegisterSignal(parent, COMSIG_HUMAN_MELEE_UNARMED_ATTACK, PROC_REF(on_punch))
+	RegisterSignal(SSdcs, COMSIG_GLOB_MOB_DEATH, PROC_REF(on_death))
+	RegisterSignal(parent, COMSIG_LIVING_RESONATOR_BURST, PROC_REF(on_resonator_burst))
+	RegisterSignal(parent, COMSIG_LIVING_PROJECTILE_PARRIED, PROC_REF(on_projectile_parry))
+	RegisterSignal(parent, COMSIG_LIVING_DEFUSED_GIBTONITE, PROC_REF(on_gibtonite_defuse))
+	RegisterSignal(parent, COMSIG_LIVING_CRUSHER_DETONATE, PROC_REF(on_crusher_detonate))
+	RegisterSignal(parent, COMSIG_LIVING_DISCOVERED_GEYSER, PROC_REF(on_geyser_discover))
+
+	projectile_parry = AddComponent(\
+		src,\
```

Comment:
er. You didn't get what i meant. You passed in src as the source argument for the sourced component part of parrying. i think it should be REF instead. this variable should still be a weakref.

Reply:
oh yeah forgot to remove src, whoops